modules/social_features/social_event/social_event.module

```diff
@@ -282,28 +284,108 @@ function _social_event_format_date($event, $view_mode) {
+      // Default time should not be displayed.
+      $end_time = $date_formatter->format($end_datetime, $time_format, '', 'UTC') === '00:01' ? '' : $date_formatter->format($end_datetime, $time_format);
+    }
+
     // Date are the same or there are no end date.
     if (empty($end_datetime) || $start_datetime == $end_datetime) {
-      $date = $date_formatter->format($start_datetime, $date_format);
-      $time = $date_formatter->format($start_datetime, $time_format);
-      $event_date = "$date - $time";
+      $event_date = empty($start_time) ? $start_date : "$start_date - $start_time";
     }
     // The date is the same, the time is different.
     elseif (date('Y-m-d', $start_datetime) == date('Y-m-d', $end_datetime)) {
-      $date = $date_formatter->format($start_datetime, $date_format);
-      $start_time = $date_formatter->format($start_datetime, $time_format);
-      $end_time = $date_formatter->format($end_datetime, $time_format);
-      $event_date = "$date $start_time - $end_time";
+      $event_date = "$start_date $start_time - $end_time";
     }
     // They are not the same day (or empty?).
     elseif (!empty($end_datetime)) {
-      $start_date = $date_formatter->format($start_datetime, $date_format);
-      $end_date = $date_formatter->format($end_datetime, $date_format);
-      $start_time = $date_formatter->format($start_datetime, $time_format);
-      $end_time = $date_formatter->format($end_datetime, $time_format);
       $event_date = "$start_date $start_time - $end_date $end_time";
     }
   }
 
   return $event_date;
 }
+
+/**
+ * Implements hook_form_alter().
+ */
+function social_event_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $event_forms = [
+    'node_event_form',
+    'node_event_edit_form',
+  ];
+
+  if (in_array($form_id, $event_forms)) {
+    $form['event_all_day'] = [
+      '#type' => 'checkbox',
+      '#title' => t('All day'),
+    ];
+    $form['#fieldgroups']['group_date_time']->children[] = 'event_all_day';
+    $form['#group_children']['event_all_day'] = 'group_date_time';
+    $form['#after_build'][] = 'social_event_date_after_build';
+  }
+}
+
+/**
+ * Add custom validation to event date fields.
+ */
+function social_event_date_after_build($form, &$form_state) {
+  array_unshift($form['field_event_date']['widget'][0]['value']['#element_validate'], 'social_event_date_validate');
+  array_unshift($form['field_event_date_end']['widget'][0]['value']['#element_validate'], 'social_event_date_validate');
+  return $form;
+}
+
+/**
+ * Set default time to the date field if time was not set.
+ */
+function social_event_date_validate(&$element, FormStateInterface $form_state, &$complete_form) {
+  $input = NestedArray::getValue($form_state->getValues(), $element['#parents']);
+  // Skip default validation for required time when date is required.
+  if (!empty($input['date']) && (empty($input['time']))) {
+    $input['time'] = '00:01:00';
+    $format = 'Y-m-d H:i:s';
```

Comment:
Not sure, but will this work when a different format is set for a date? I know that one client uses a different format for event dates.

Reply:
I'm not sure if this is used for display or calculating. Display should definitely use an existing format.